src/Entity/Cliente.php

```diff
@@ -54,24 +60,29 @@ public function setCpf(string $cpf): static
         return $this;
     }
 
-    public function getCarrinho(): ?Carrinho
+    public function getCarrinhos(): Collection
     {
-        return $this->carrinho;
+        return $this->carrinhos;
     }
 
-    public function setCarrinho(?Carrinho $carrinho): static
+    public function addCarrinho(Carrinho $carrinho): static
     {
-        // unset the owning side of the relation if necessary
-        if ($carrinho === null && $this->carrinho !== null) {
-            $this->carrinho->setCliente(null);
-        }
-
-        // set the owning side of the relation if necessary
-        if ($carrinho !== null && $carrinho->getCliente() !== $this) {
+        if (!$this->carrinhos->contains($carrinho)) {
+            $this->carrinhos[] = $carrinho;
             $carrinho->setCliente($this);
         }
 
-        $this->carrinho = $carrinho;
+        return $this;
+    }
+
+    public function removeCarrinho(Carrinho $carrinho): static
+    {
+        if ($this->carrinhos->contains($carrinho)) {
+            $this->carrinhos->removeElement($carrinho);
+            if ($carrinho->getCliente() === $this) {
+                $carrinho->setCliente(null);
+            }
+        }
```

Comment:
já vai ter removido então não precisa do setCliente...

```suggestion
        if ($this->carrinhos->contains($carrinho)) {
            $this->carrinhos->removeElement($carrinho);
        }
```

Reply:
ajusta aqui @GabrielCirqueira